### PR TITLE
tests/libcose: Move from unittests to regular test

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

Moves the libcose unittest to a "regular" test in `pkg_libcose`. saves quite a bit of flash and ram on a samr21-xpro for the unittests (No, still doesn't fit):
Before:
```
region `rom' overflowed by 159324 bytes
region `ram' overflowed by 17180 bytes
```
After:
```
region `rom' overflowed by 131664 bytes
region `ram' overflowed by 12100 bytes
```
### Testing procedure

Run the test in pkg_libcose.

### Issues/PRs references

None

### CODE REVIEW

**Comment by miri64** [comment]
:laughing: to

> (No, still doesn't fit)

Not the PR itself ;-)

File: tests/pkg_libcose/Makefile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+include ../Makefile.tests_common
+
+# HACL* only compiles on 32bit platforms
+BOARD_BLACKLIST := arduino-duemilanove arduino-mega2560 arduino-uno \
+                   jiminy-mega256rfr2 mega-xplained waspmote-pro \
+                   chronos msb-430 msb-430h telosb wsn430-v1_3b wsn430-v1_4 z1
+
+TEST_ON_CI_WHITELIST += all
+
+CFLAGS += -DTHREAD_STACKSIZE_MAIN=\(4*THREAD_STACKSIZE_DEFAULT\)
```
Comment (miri64) [inline tests/pkg_libcose/Makefile]: Might the stacksize in the unittests be reduced then maybe?

Comment (bergzand) [reply tests/pkg_libcose/Makefile]: I'm afraid not, looking at [this] commit, the stacksize required this size because of the tweetnacl unittest. That commit is from before libcose was included in the unittests.

[this]: https://github.com/RIOT-OS/RIOT/commit/809c4b772450c2261462b31db3d6c99ecdde1eec

Comment (miri64) [reply tests/pkg_libcose/Makefile]: Ok. Maybe we should move `tweetnacl` out of the unittests as well (but this is OOSOTPR (out-of-scop of this PR)

**Comment by bergzand** [comment]
Should be fixed now

File: tests/pkg_libcose/tests/01-run.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+import sys
+from testrunner import run
+
+
+def testfunc(child):
+    child.expect_exact('OK (3 tests)')
```
Comment (miri64) [inline tests/pkg_libcose/tests/01-run.py]: Maybe make this a regex. This way the test is extensible (and might not fail randomly ;-))

```suggestion
    child.expecr(r'OK (\d+ tests)')
```

Comment (bergzand) [reply tests/pkg_libcose/tests/01-run.py]: Agreed, Fixed

**Comment by bergzand** [comment]
I think I have to increase the timeout on the CI run test.

**Comment by bergzand** [comment]
> Seems saclay is currently down :-/.

Ah, that means it's not only down for me.

Testing it locally shows that the test takes around 67 seconds:
```
2018-10-17 22:28:20,717 - INFO # main(): This is RIOT! (Version: 2018.10-devel-1213-g386c9-dystopia-pr/tests/libcose_move)
2018-10-17 22:29:27,138 - INFO # ...
2018-10-17 22:29:27,140 - INFO # OK (3 tests)
```

Do we actually want to have this run on the CI if it takes this long?

**Comment by bergzand** [comment]
Increased the timeout to 120s

**Comment by miri64** [comment]
> Do we actually want to have this run on the CI if it takes this long?

Good question. IMHO for nightly builds this should be done, but maybe not for every PR that sets `CI: run tests`. I don't know (except with re-introducing not running/compiling uneffected test) this is possible right now.

**Comment by bergzand** [comment]
@kaspar030 @smlng Do you guys have an opinion on this?

**Comment by miri64** [comment]
An answer to our question would be nice though ;-)

**Comment by smlng** [comment]
I like this, see #10188 and #10189 😉 IIRC I had a similar discussion with others (e.g. @cladmi) to finally have packages tests within the package directory and make them self-contained. So moving the tests out of unittests is a first step in that direction and also has the benefit of reducing unittests binary size.

TL;DR: 👍 🎉  ❤️

**Comment by miri64** [comment]
But the question was if we want to introduce a **2min** wait-time to the CI tests, because this test requires two minutes to finish on samr21-xpro.

**Comment by smlng** [comment]
> But the question was if we want to introduce a **2min** wait-time to the CI tests, because this test requires two minutes to finish on samr21-xpro.

ah that question: well, I would say no don't run the tests on hardware if it takes that long

File: tests/pkg_libcose/Makefile [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+# HACL* only compiles on 32bit platforms
+BOARD_BLACKLIST := arduino-duemilanove arduino-mega2560 arduino-uno \
+                   jiminy-mega256rfr2 mega-xplained waspmote-pro \
+                   chronos msb-430 msb-430h telosb wsn430-v1_3b wsn430-v1_4 z1
+
+BOARD_INSUFFICIENT_MEMORY := nucleo-f030r8 nucleo-f031k6 nucleo-f042k6 \
+                             nucleo-l031k6 nucleo-l053r8 stm32f0discovery
+
+TEST_ON_CI_WHITELIST += all
```
Comment (miri64) [inline tests/pkg_libcose/Makefile]: > > But the question was if we want to introduce a **2min** wait-time to the CI tests, because this test requires two minutes to finish on samr21-xpro.
> 
> ah that question: well, I would say no don't run the tests on hardware if it takes that long

@bergzand then please change this

```suggestion
TEST_ON_CI_WHITELIST += native
```

Comment (bergzand) [reply tests/pkg_libcose/Makefile]: Good idea! Applied

**Comment by bergzand** [comment]
I've extended the test script a bit to only use the 120 seconds timeout when running on non-native. Now when the test breaks, the CI doesn't have to wait for the 2 minute timeout to pass on native.

**Comment by cladmi** [comment]
@smlng we discussed this IRL a few month ago with @bergzand and he proposed to move them out of `unittest` as it will simplify moving them to `pkg` later. And it is indeed a first step in that direction.

**Comment by bergzand** [comment]
squashed!

**Comment by bergzand** [comment]
Ready to merge here?

**Comment by miri64** [comment]
Yepp

**Comment by bergzand** [comment]
Thanks for reviewing!
